### PR TITLE
Create cf-782b

### PR DESCRIPTION
**Pull request by EulerPascal404**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/cf-782b
Comment (SansPapyrus683) [inline solutions/cf-782b:35]: is this under 80 chars?

File: solutions/cf-782b
Comment (SansPapyrus683) [inline solutions/cf-782b:29]:
```suggestion
double solve(double loc) {
```

File: solutions/cf-782b
Comment (SansPapyrus683) [inline solutions/cf-782b:82]: if your comments are this long, just put it in an `Explanation` header above

File: solutions/cf-782b
Comment (SansPapyrus683) [inline solutions/cf-782b:93]: same here

File: solutions/cf-782b
Comment (SansPapyrus683) [inline solutions/cf-782b:102]:
```suggestion

	cout << setprecision(13) << min_ans << endl;
```

File: solutions/cf-782b
Comment (SansPapyrus683) [inline solutions/cf-782b:62]: sus

File: solutions/cf-782b
Comment (SansPapyrus683) [inline solutions/cf-782b:56]:
```suggestion
	return curr_ans;
```
comments need not state the obvious

File: solutions/cf-782b
Comment (SansPapyrus683) [inline solutions/cf-782b:42]:
```suggestion
			} else {
```

File: solutions/cf-782b
Comment (SansPapyrus683) [inline solutions/cf-782b:38]: two of the same comment?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cf-782b
Comment (EulerPascal404) [reply solutions/cf-782b:82]: What do you mean by `Explanation` header?

**Comment by EulerPascal404** [comment]
@SansPapyrus683 Just made all the requested changes.

File: solutions/cf-782b
Comment (SansPapyrus683) [inline solutions/cf-782b:33]: assignments on separate lines

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/cf-782b
Comment (EulerPascal404) [reply solutions/cf-782b]: done

**Comment by EulerPascal404** [comment]
@SansPapyrus683 I think this can also be merged. Except if I am still missing something.

**Comment by SansPapyrus683** [comment]
let's wait for 1 more approval

**Comment by EulerPascal404** [comment]
> let's wait for 1 more approval

alright.

**Comment by EulerPascal404** [comment]
Oops, that was an accident.

File: solutions/cf-782b
Comment (envyaims) [inline solutions/cf-782b:83]: first condition isnt necessary and make the error a constant

File: solutions/cf-782b
Comment (envyaims) [inline solutions/cf-782b:36]: rename the function to something better, like get_time or something

File: solutions/cf-782b
Comment (envyaims) [inline solutions/cf-782b:96]:
```suggestion
	cout << fixed << min_ans << endl;
```
isnt error just 1e-6?

File: solutions/cf-782b
Comment (envyaims) [inline solutions/cf-782b:33]: put the comment on top

File: solutions/cf-782b
Comment (envyaims) [inline solutions/cf-782b:37]: wtf you already declared ```curr_ans``` as double up there

also rename ```curr_ans``` to something better, like ```max_time```

File: solutions/cf-782b
Comment (envyaims) [inline solutions/cf-782b:54]:
```suggestion
			// The max can be from both sides, so we don't reset the 2 booleans
```

File: solutions/cf-782b
Comment (envyaims) [inline solutions/cf-782b:85]:
```suggestion
		curr_ans = solve(mid);  // returns time it would take if friends were to meet at mid
```

**Review by envyaims — CHANGES_REQUESTED** [review]

**Comment by envyaims** [comment]
also you forgot .mdx

File: solutions/cf-782b
Comment (SansPapyrus683) [reply solutions/cf-782b:33]: (optional)

File: solutions/cf-782b
Comment (SansPapyrus683) [reply solutions/cf-782b:36]: same dude who thought `dfs` was an acceptable function name

**Comment by EulerPascal404** [comment]
@envyaims I think I applied all your revisions.

**Review by envyaims — APPROVED** [review]
looks neato burrito

File: solutions/cf-782b.mdx
Comment (ryanchou-dev) [inline solutions/cf-782b.mdx:14]:
```suggestion
Whenever we are calculating the answer, we take note of which side the maximum answer came from. If the max answer comes from both sides, regardless of where we decide to go next, our answer will only increase, so we return that answer. Otherwise, if it only comes from one side of the point that we just calculated, we move our search towards that side, as moving it towards the other side will only increase our answer. 
```

**Review by ryanchou-dev — APPROVED** [review]
okk

**Pull request merged**
